src/core/annotation.js

```diff
@@ -1066,14 +1066,21 @@ class WidgetAnnotation extends Annotation {
     }
 
     let loopDict = dict;
+    const visited = new Set([this.data.id]);
     while (loopDict.has("Parent")) {
       loopDict = loopDict.get("Parent");
-      if (!isDict(loopDict)) {
+      if (!isDict(loopDict) || visited.has(loopDict.objId)) {
```

Comment:
```suggestion
      if (
        !(loopDict instanceof Dict) ||
        (loopDict.objId && visited.has(loopDict.objId))
      ) {
```

Reply:
It won't work if `objId` is null which should be the case if the parent is inside the object (no ref).